topology/graph/indexer.go

```diff
@@ -28,128 +28,180 @@ import (
 	"github.com/cnf/structhash"
 )
 
-type NodeHasher func(n *Node) string
+// NodeHasher describes a callback that is called to map a node to
+// a set of hash,value pairs
+type NodeHasher func(n *Node) map[string]interface{}
 
+// GraphIndexer describes
```

Comment:
can you elaborate :)

Reply:
At least, it leaves room for imagination :-)